src/ucp/core/ucp_listener.c

```diff
@@ -523,6 +523,12 @@ void ucp_listener_destroy(ucp_listener_h listener)
 {
     ucs_trace("listener %p: destroying", listener);
 
+    UCS_ASYNC_BLOCK(&listener->worker->async);
+    ucs_callbackq_remove_if(&listener->worker->uct->progress_q,
```

Comment:
i must be missing something here, where is this function added to the queue?
ucp_cm_server_conn_request_progress_cb_pred()

Reply:
AFAIU `ucp_cm_server_conn_request_progress_cb_pred` is a predicate callback to remove `ucp_cm_server_conn_request_progress` from the queue and reject conn request

Reply:
@alinask it's added in ucp_cm_server_conn_request_cb but not removed in ucp_listener_destroy

Reply:
@yosefe can you please point to where it's added in ucp_cm_server_conn_request_cb?

Reply:
https://github.com/openucx/ucx/blob/c786a066352eb6b493cb46de53f490531c75ef11/src/ucp/wireup/wireup_cm.c#L952

Reply:
thanks but meant where ucp_cm_server_conn_request_progress_cb_pred() is added.
it's kind of an 'extension' of ucp_cm_server_conn_request_progress?

Reply:
ucp_cm_server_conn_request_progress_cb_pred is not added, it's just helper function for ucs_callbackq_remove_if